### PR TITLE
Add keepUnicode value in the cache key of normalizedPathCache

### PR DESCRIPTION
**Pull request by PVince81**
Please review @owncloud/filesystem @nickvergessen @georgehrke

### CODE REVIEW

**Comment by georgehrke** [comment]
What issue does this fix?
(Or how to properly test this?)

**Comment by PVince81** [comment]
Good point. Now thinking of it this should be testable.

Basically if you call `normalizePath("somepath")` once without `keepUnicode` and once with `keepUnicode`, the cache would give you the same result even though it's wrong.

I'll add a test.

**Comment by nickvergessen** [comment]
I hope its using cappedcache? :stuck_out_tongue_winking_eye: 

:+1: looks save and shouldn't break anything, apart from wrong cache hits.
So yes, adding a test would be nice

**Comment by PVince81** [comment]
@nickvergessen yes, there's a capped cache

I need your help here:
```
		$this->invokePrivate('\OC\Files\Filesystem', '$normalizedPathCache', [null]);
```

How can I clear the private cache attribute ? (it's on my local branch)

**Comment by nickvergessen** [comment]
```php
$this->invokePrivate('\OC\Files\Filesystem', 'normalizedPathCache', [null]);
```

No dollar-sign needed? the name of the member is `normalizedPathCache` (you use it with `$this->normalizedPathCache` and not `$this->$normalizedPathCache` ;) )

**Comment by PVince81** [comment]
Okay, now got this:
```
get_class() expects parameter 1 to be object, string given
```

**Comment by PVince81** [comment]
Probably doesn't work with static classes :frowning:
Maybe I'd just add a new method `clearNormalizedPathCache()` and call that in `tearDown`

**Comment by PVince81** [comment]
Okay, I managed to fix `invokePrivate` and I removed the `get_class`.

Will push this and let's see if it kills other tests...

File: tests/lib/TestCase.php [hunk 1]
```diff
@@ -166,7 +166,7 @@ protected function tearDown() {
 	 * @return mixed
 	 */
 	protected static function invokePrivate($object, $methodName, array $parameters = array()) {
-		$reflection = new \ReflectionClass(get_class($object));
+		$reflection = new \ReflectionClass($object);
```
Comment (nickvergessen) [inline tests/lib/TestCase.php]: needs a bit more work so it takes both, objects and strings

**Comment by PVince81** [comment]
@nickvergessen this should work now: https://github.com/owncloud/core/pull/24940/commits/5b1eb416d87c0e85e60547003f9b01847c81ec8a

Needed to use a separate variable because the original value of `$object` is used later on.

**Comment by PVince81** [comment]
Tests pass now.

@georgehrke @nickvergessen

**Comment by nickvergessen** [comment]
I'd like to get 7 pass, just to be sure the static Reflection works there as well

**Comment by nickvergessen** [comment]
Will test locally

**Comment by nickvergessen** [comment]
```
~/vendor/bin/phpunit --configuration phpunit-autotest.xml --coverage-clover autotest-clover-sqlite.xml --coverage-html coverage-html-sqlite --log-junit autotest-results-sqlite.xml ../tests/lib/Files/FilesystemTest.php 
PHPUnit 5.3.4 by Sebastian Bergmann and contributors.

Runtime:       PHP 7.0.7-2+donate.sury.org~trusty+1
Error:         No code coverage driver is available

...............................................................  63 / 109 ( 57%)
.............................SSSSSS...........                  109 / 109 (100%)

Time: 644 ms, Memory: 12.00MB

OK, but incomplete, skipped, or risky tests!
Tests: 109, Assertions: 231, Skipped: 6.
```

:+1:

**Comment by icewind1991** [comment]
:+1: looks good

**Pull request merged**